### ISSUE TITLE
utf-8 invalid start byte error (û error)

### ISSUE THREAD

**Issue by PatGendre** [e-mission/e-mission-docs#458]
we encounter this kind of error repeatedly in the webserver log. 
0xfb is û like août/French August, so there is still an encoding problem somewhere.

Maybe due only to the older usercache data that I need to clean? 
or maybe due to some application which has not been updated with the patch correcting the utf-8 issue : how to find the user who needs to update the app? I have to find the UUID from the long "user" value in the first field of the body sent?

`START POST /usercache/put
2019-09-17 00:55:28,437:DEBUG:139931544434432:Called userCache.put
2019-09-17 00:55:28,437:DEBUG:139931544434432:@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@ JSON
2019-09-17 00:55:28,437:DEBUG:139931544434432:ctype = application/json
2019-09-17 00:55:28,476:DEBUG:139931544434432:body = b'{...
... a very long JSON (3 MB)
...
2019-09-17 01:55:12,017:DEBUG:139931860272896:'utf-8' codec can't decode byte 0xfb in position 1993728: invalid start byte
2019-09-17 01:55:12,017:DEBUG:139931860272896:@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@ HTTPError`

**Comment by shankari** [comment]
> Maybe it's due only to the older usercache data that I need to clean?

This is not due to older usercache data; this is logged as part of the `userCache.put` API call, which is the one invoked to upload new data to the server.

```
2019-09-17 00:55:28,437:DEBUG:139931544434432:@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@ JSON 2019-09-17 00:55:28,437:DEBUG:139931544434432:ctype = application/json 2019-09-17 00:55:28,476:DEBUG:139931544434432:body = b'{... ... a very long JSON (3 MB) ... 2019-09-17 01:55:12,017:DEBUG:139931860272896:'utf-8' codec can't decode byte 0xfb in position 1993728: invalid start byte 2019-09-17 01:55:12,017:DEBUG:139931860272896:@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@ HTTPError
```

These are the additional logs added by Yann when we were debugging the prior UTF-8 error. This seems to indicate that there are some clients which are still sending non-UTF-8 data to the server. 

Can you see which users are affected? Are you seeing the same messages for your account?

**Comment by PatGendre** [comment]
> Can you see which users are affected? 
I don't see where the uuid is in the logs; in the JSON body after the usercache put call, it is encoded as "user":"eyJhbGciOiJSUzI1NiIsInR5cCIgOiAiSldUIiwia2lkIiA6ICJmTHFhVGFZMzdqeTZOSVhpbVU2TUZUTGdXa1NTUFpUck1xVjQ4RmZ3czVjIn0.eyJqdGkiOiIyMGI0OGQ1YS1hNzA3LTRmNTEtOTY0ZC0wNjI0MDg0YjY0NDQiLCJleHAiOjE1Njg2NzQ3OTEsIm5iZiI6MCwiaWF0IjoxNTY4Njc0NDkxLCJpc3MiOiJodHRwczovL2lkLmluZGllLmhvc3QvYXV0aC9yZWFsbXMvZmFibW9iLWlvIiwiYXVkIjoidHJhY2V1ciIsInN1YiI6Ijc3OTBjYTE0LTNhYmUtNGJjMi1iMTIzLTc4NjBkNWRkNmVhOCIsInR5cCI6IklEIiwiYXpwIjoidHJhY2V1ciIsImF1dGhfdGltZSI6MTU2Mzc4NzE0Niwic2Vzc2lvbl9zdGF0ZSI6ImM3YzMzODFiLTk1OGQtNDJjMy04ZThhLTMwZTRiNGYwMzY2MyIsImFjciI6IjEiLCJlbWFpbF92ZXJpZmllZCI6ZmFsc2UsIm5hbWUiOiJmcmVkZXJpYyBsYWl0aGllciIsInByZWZlcnJlZF91c2VybmFtZSI6ImZyZWRlcmljLmwiLCJnaXZlbl9uYW1lIjoiZnJlZGVyaWMiLCJmYW1pbHlfbmFtZSI6ImxhaXRoaWVyIiwiZW1haWwiOiJmcmVkZXJpYy5sYWl0aGllckBmbG93YmlyZC5ncm91cCJ9.Ys9fgfF5c_BaqNj_riJODfkblHt06zNYE62BZVz2QUGYUjdc0To_O9ueUhgmw5jDMAGaJr1ZxnatD87KtILv4xUtU_eIqUk2GpUeXr_tM-rqW6DawWHQfOcej3JUZy73E_ySrM3IdPnZtu06dRnVDP_pY9AoJfONJKUR0lxP7Kj0Kwm9Utrx8hSoDtAQ55LaqTos-bdIiH63QwFpeg2qp5t5QA1QJ6NAfMhTDnNerVUUXTivj-PHE6fCS_twlGMpvq6phjCMXChYRIUomRwryVHWS1_6xyQ90iLkqQ6YfdddawKn0QKI-YByM-oi3uXvwH88cxNZkbRTcLr7mqK6Og"

**Comment by PatGendre** [comment]
good news, may be the user has updated the app?
in any case for the last 2 days I don't see the http error in the logs any more :-)

**Comment by shankari** [comment]
> I don't see where the uuid is in the logs; in the JSON body after the usercache put call, it is encoded as

That is the user token; you would need to decode it using the keys from the OpenID server, and this is non-trivial. You would need to look for subsequent logs that decode the token - e.g. around `After verifying claim, returning valid email` or at the UUID reported as part of the END call - e.g.

```
START 2019-09-18 22:56:25.185973 POST /usercache/put
END 2019-09-18 22:56:25.396647 POST /usercache/put adb1b826-e887-46ed-89b2-747dd5568735 0.2106008529663086
```

In this case, `adb1b826-e887-46ed-89b2-747dd5568735` is the UUID.

**Comment by PatGendre** [comment]
Yes, the uuid seems to be an empty string? Just after the httpError i have this :
`2019-09-17 00:55:28,498:DEBUG:139931544434432:@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@ HTTPError
2019-09-17 00:55:28,499:ERROR:139931544434432:old style duration 31.177546739578247 != timer based duration 0.06232224125415087
2019-09-17 00:55:28,501:DEBUG:139931544434432:insert called with entry of type <class 'emission.core.wrapper.entry.Entry'>
2019-09-17 00:55:28,502:DEBUG:139931544434432:entry was fine, no need to fix it
2019-09-17 00:55:28,502:DEBUG:139931544434432:Inserting entry Entry({'_id': ObjectId('5d8012e04e276d46bd8698b0'), 'user_id': '', 'metadata': Metadata({'key': 'stats/server_api_error', 'platform': 'server', 'write_ts': 1568674528.500924, 'time_zone': 'America/Los_Angeles', 'write_local_dt': LocalDate({'year': 2019, 'month': 9, 'day': 16, 'hour': 15, 'minute': 55, 'second': 28, 'weekday': 0, 'timezone': 'America/Los_Angeles'}), 'write_fmt_time': '2019-09-16T15:55:28.500924-07:00'}), 'data': {'name': 'MISMATCH_POST_/usercache/put', 'ts': 1568674528.4991884, 'reading': 31.115224498324096}}) into timeseries
2019-09-17 00:55:28,504:DEBUG:139931544434432:END POST /usercache/put  31.177546739578247`
